tests/Moq.Tests/InvocationShapeFixture.cs

```diff
@@ -0,0 +1,63 @@
+// Copyright (c) 2007, Clarius Consulting, Manas Technology Solutions, InSTEDD.
+// All rights reserved. Licensed under the BSD 3-Clause License; see License.txt.
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Linq.Expressions;
+using System.Reflection;
+
+using Xunit;
+
+namespace Moq.Tests
+{
+	public class InvocationShapeFixture
+	{
+		[Fact]
+		public void Regular_parameters_are_compared_using_equality()
+		{
+			var (expr1, method1, args1) = Deconstruct<A>(a => a.Method(1, 2, 3));
+			var (expr2, method2, args2) = Deconstruct<A>(a => a.Method(1, 2, 3));
+
+			Assert.NotSame(expr1.Body, expr2.Body);  // let's make sure the compiler didn't optimize the above
+
+			var expectation1 = new InvocationShape(expr1, method1, args1);
+			var expectation2 = new InvocationShape(expr2, method2, args2);
+
+			Assert.Equal(expectation1, expectation2);
+		}
+
+		[Fact]
+		public void Param_array_args_are_compared_using_structural_equality_not_reference_equality()
+		{
+			var (expr1, method1, args1) = Deconstruct<B>(b => b.Method(1, 2, 3));
+			var (expr2, method2, args2) = Deconstruct<B>(b => b.Method(1, 2, 3));
+
+			Assert.NotSame(expr1.Body, expr2.Body);  // let's make sure the compiler didn't optimize the above
+
+			var expectation1 = new InvocationShape(expr1, method1, args1);
+			var expectation2 = new InvocationShape(expr2, method2, args2);
+
+			Assert.Equal(expectation1, expectation2);
+		}
```

Comment:
❗️ These two tests are almost identical. Perhaps extract what's common to both into a private method.